docs/backends.rst

```diff
@@ -0,0 +1,211 @@
+===================================
+ Proctoring backend implementation
+===================================
+
+Proctoring providers (PS) who wish to integrate with Open edX should implement a `REST API`_ and a thin `Python wrapper`_, as described below.
+
+REST API
+--------
+
+Implement the following endpoints. In order to authenticate requests from the Open edX server, the PS backend should
+enable Oauth 2 authentication using JWT_. The Open edX installation will be configured with a client key and a client secret provided by the PS and will authenticate each request to the PS.
+
+To obtain a JWT_ token, the PS system must make an Oauth2 request to ``/oauth2/access_token``.
+
+SSL is required for all (production environment) requests.
+
+All requests and responses in this API are formatted as JSON objects.
+
+
+Proctoring System configuration endpoint
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    /v1/config/
+
+``GET``: returns an object of the available configuration options and metadata for the proctoring service.::
+
+    {
+        "config": {
+            "allow_multiple": "Allow multiple monitors",
+            "allow_notes": "Allow paper notes",
+            "allow_apps": "Allow other applications to be running",
```

Comment:
As written each of these assume that all rule configurations are boolean values. Can we
1) minimally, document the fact that our api assumes all rules can be expressed as booleans, xor
2) complicate this slightly to e.g. `"allow_apps": {"name": "Allow other applications to be running", "type": "boolean" },` or something else which enables more rule configuration types in the future?

Reply:
I agree that settings may need to be more flexible in the future. I suggest that you take this task on after this PR has been merged. That would involve changing this doc, the proctoring api, and the mockprock server.

Reply:
Can you capture it in a ticket/subtask of something, then, to help ensure it doesn't get lost in the mail?